aws/resource_aws_guardduty_publishing_destination_test.go

```diff
@@ -0,0 +1,273 @@
+		CheckDestroy: testAccCheckAwsGuardDutyPublishingDestinationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsGuardDutyPublishingDestinationConfig_basic(bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsGuardDutyPublishingDestinationExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "detector_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "destination_arn"),
+					resource.TestCheckResourceAttrSet(resourceName, "kms_key_arn"),
+					resource.TestCheckResourceAttr(resourceName, "destination_type", "S3")),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+const testAccGuardDutyPublishingDestinationConfig_basic1 = `
+
+data "aws_caller_identity" "current" {}
+
+data "aws_region" "current" {}
+
+data "aws_iam_policy_document" "bucket_pol" {
+  statement {
+    sid = "Allow PutObject"
+    actions = [
+      "s3:PutObject"
+    ]
+
+    resources = [
+      "${aws_s3_bucket.gd_bucket.arn}/*"
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["guardduty.amazonaws.com"]
+    }
+  }
+
+  statement {
+    sid = "Allow GetBucketLocation"
+    actions = [
+      "s3:GetBucketLocation"                                                   
+    ]
+
+    resources = [
+      "${aws_s3_bucket.gd_bucket.arn}"
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["guardduty.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "kms_pol" {
+
+  statement {
+    sid = "Allow GuardDuty to encrypt findings"
+    actions = [
+      "kms:GenerateDataKey"
+    ]
+
+    resources = [
+      "arn:aws:kms:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:key/*"
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["guardduty.amazonaws.com"]
+    }
+  }
+
+  statement {
+    sid = "Allow all users to modify/delete key (test only)"
+    actions = [
+      "kms:*"
+    ]
+
+    resources = [
+      "arn:aws:kms:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:key/*"
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers = [ "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root" ]
+    }
+  }
+
+}
+
+resource "aws_guardduty_detector" "test_gd" {
+  enable = true
+}
+
+resource "aws_s3_bucket" "gd_bucket" {
+  bucket = "<<BUCKET_NAME>>"
+  acl    = "private"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_policy" "gd_bucket_policy" {
+  bucket = aws_s3_bucket.gd_bucket.id
+  policy = data.aws_iam_policy_document.bucket_pol.json
+}
+
+resource "aws_kms_key" "gd_key" {
+  description = "Temporary key for AccTest of TF"
+  deletion_window_in_days = 7
+  policy = data.aws_iam_policy_document.kms_pol.json
+}`
+
+func testAccAwsGuardDutyPublishingDestinationConfig_basic(bucketName string) string {
+	return fmt.Sprintf(`
+	%[1]s
```

Comment:
can we combine these? it's not being reused.

Reply:
Refactored